develop/tutorials/articles/140-application-security/01-application-permissions/05-check-permissions.markdown

```diff
@@ -1,3 +1,240 @@
+                int displayDateHour, int displayDateMinute, boolean allowPingbacks,
+                boolean allowTrackbacks, String[] trackbacks,
+                String coverImageCaption, ImageSelector coverImageImageSelector,
+                ImageSelector smallImageImageSelector,
+                ServiceContext serviceContext)
+            throws PortalException {
+
+            _portletResourcePermission.check(
+                getPermissionChecker(), serviceContext.getScopeGroupId(),
+                ActionKeys.ADD_ENTRY);
+
+            return blogsEntryLocalService.addEntry(
+                getUserId(), title, subtitle, description, content,
+                displayDateMonth, displayDateDay, displayDateYear, displayDateHour,
+                displayDateMinute, allowPingbacks, allowTrackbacks, trackbacks,
+                coverImageCaption, coverImageImageSelector, smallImageImageSelector,
+                serviceContext);
+        }
+
+    The check throws an exception if it fails, preventing the local service call
+    that adds the entry. A convention Liferay uses is to place the action keys
+    from `default.xml` as constants in an `ActionKeys` class. If `ActionKeys`
+    doesn't have an action key appropriate for your application, extend
+    Liferay's class and add your own keys. 
+
+Add permission checks where necessary to protect your application's functions at
+the service level. Next, you'll learn how to create permission helper classes
+for your web module. 
+
+## Create Permission Helper Classes in Your Web Module
+
+A helper class can make it easier to check permissions in your portlet
+application. You can create helper classes for both portlet permissions and for
+model permissions. Here's how to create a portlet permission helper: 
+
+1.  Create a package with the suffix
+    `web.internal.security.permission.resource`. For example, the Blogs
+    application has the package
+    `com.liferay.blogs.web.internal.security.permission.resource`. 
+
+2.  Create a component class with at least one static method for checking
+    permissions. For example, here's the `BlogsPermission` class: 
+
+        @Component(immediate = true)
+        public class BlogsPermission {
+
+            public static boolean contains(
+                PermissionChecker permissionChecker, long groupId, String actionId) {
+
+                return _portletResourcePermission.contains(
+                    permissionChecker, groupId, actionId);
+            }
+
+            @Reference(
+                target = "(resource.name=" + BlogsConstants.RESOURCE_NAME + ")",
+                unbind = "-"
+            )
+            protected void setPortletResourcePermission(
+                PortletResourcePermission portletResourcePermission) {
+
+                _portletResourcePermission = portletResourcePermission;
+            }
+
+            private static PortletResourcePermission _portletResourcePermission;
```

Comment:
Note the field is static while the setter method is an instance method.  This is how we've chosen to get around having service references in jsps.